util/encoding.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+// Helpers functions regarding the encoding of basic types ( integers ...)
```

Comment:
Would it be okay then if I implement methods like "UInt64ToBuf" and "BufToUInt64", but still stay with the same "int" in my structs like in PartialSchnorrSig.Index ? I would just transform the Index into a uint64 before passing it to UInt64ToBuf. That way, the binary format would know we use a "uint64" but the structs still use the same good old "int"